lib/registrations.js

```diff
@@ -124,15 +134,96 @@ const asClass = (Type, opts) => {
 
   opts = makeOptions(defaults, opts)
 
-  const resolve = (container) => new Type(container.cradle)
-
+  const resolve = generateResolve(buildNewClassFunction(Type), Type.prototype.constructor)
   const result = {
     resolve,
-    lifetime: opts.lifetime
+    lifetime: opts.lifetime,
+    resolutionMode: opts.resolutionMode
   }
-
+  result.resolve = resolve.bind(result)
   Object.assign(result, makeFluidInterface(result))
+
   return result
 }
 
+/**
+ * Returns a function used to invoke a type constructor.
+ *
+ * @param {Class} Type
+ * The type being contructed.
+ *
+ * @return {Function}
+ * The function that constructs the specified type
+ */
+function buildNewClassFunction (Type) {
+  return (dependencies) => {
+    if (!dependencies) {
+      return new Type()
+    } else if (dependencies instanceof Array) {
+      return new Type(...dependencies)
+    }
+
+    return new Type(dependencies)
+  }
+}
+
+/**
+ * Returns a resolve function used to construct the dependency graph
+ *
+ * @param {Function} fn
+ * The function to construct
+ *
+ * @param {Function} depsFn
+ * The function to parse for the dependencies of the construction target
+ *
+ * @param {boolean} isFunction
+ * Is the resolution target an actual function or a mask for a constructor?
+ *
+ * @return {Function}
+ * The function used for dependency resolution
+ */
+function generateResolve (fn, depsFn, isFunction) {
+  // Use a regular function instead of an arrow function to facilitate binding to the container object
+  return function (container) {
+    // Because the container holds a global reolutionMode we need to determine it in the proper order or precedence:
+    // registration -> container -> default value
+    let resolutionMode = (this.resolutionMode || container.__resolutionMode || ResolutionModes.PROXY)
```

Comment:
I agree with having `options` on the container. I was thinking along those lines but you weren't storing the options on the container before so I didn't know if that would be too much. The `__` was intentional so hopefully no one would use the property haha.

I will make the other changes as well.